File: src/main/java/com/procurement/notice/model/ocds/Option.java
package com.procurement.notice.model.ocds;

import com.fasterxml.jackson.annotation.*;
import lombok.Getter;

@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "hasOptions",
        "optionDetails"
})
public class Option {
    @JsonProperty("hasOptions")
    @JsonPropertyDescription("A True/False field to indicate if lot options will be accepted. Required by the EU")
    private final Boolean hasOptions;

    @JsonProperty("optionDetails")
    @JsonPropertyDescription("Further information about the lot options that will be accepted. Required by the EU")
    private final String optionDetails;

    @JsonCreator
    public Option(@JsonProperty("hasOptions") final Boolean hasOptions,
                  @JsonProperty("optionDetails") final String optionDetails) {
        this.hasOptions = hasOptions;
        this.optionDetails = optionDetails;
    }
}

File: src/main/kotlin/com/procurement/notice/model/ocds/Option.kt
package com.procurement.notice.model.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

data class Option @JsonCreator constructor(

    @get:JsonProperty("hasOptions")
    @get:JsonInclude(JsonInclude.Include.NON_NULL)
    val hasOptions: Boolean?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val optionDetails: String?,

    @JsonInclude(JsonInclude.Include.NON_NULL)
    @field:JsonProperty("description") @param:JsonProperty("description") val description: String?,

    @JsonInclude(JsonInclude.Include.NON_NULL)
    @field:JsonProperty("period") @param:JsonProperty("period") val period: Period?
)
